molecule/default/roles/helm/tasks/test_up_dep.yml

```diff
@@ -0,0 +1,175 @@
+# Helm module
+- name: copy chart
+  copy:
+    src: "{{ item }}"
+    dest: /tmp
+  loop:
+    - test-chart
+    - dep-up
+
+- name: "Test chart with dependency_update false"
+  block:
+    - name: "Test chart with dependency_update false"
+      helm:
+        binary_path: "{{ helm_binary }}"
+        name: test
+        chart_ref: "/tmp/test-chart"
+        chart_version: "{{ chart_source_version | default(omit) }}"
+        namespace: "{{ helm_namespace }}"
+        dependency_update: false
+        create_namespace: yes
+      register: release
+
+    - name: "Check if the subchart exist in chart"
+      stat:
+        path: "/tmp/test-chart/Chart.lock"
+      register: stat_result
+
+    - assert:
+        that:
+          - not stat_result.stat.exists
+        success_msg: "subchart not exist in the chart directory"
+        fail_msg: "subchart exist in the charts directory"
+
+- name: "Test chart without dependencies block and dependency_update true"
+  block:
+    - name: "Test chart without dependencies block"
+      helm:
+        binary_path: "{{ helm_binary }}"
+        name: test
+        chart_ref: "/tmp/test-chart"
+        chart_version: "{{ chart_source_version | default(omit) }}"
+        namespace: "{{ helm_namespace }}"
+        create_namespace: yes
+        dependency_update: true
+      ignore_errors: true
+      register: release
+
+    - assert:
+        that:
+          - release.failed
+          - release.msg == "No subchart will be pulled for /tmp/test-chart. Please make sure to add dependencies block in Chart.yaml or requirements.yaml. For more information please visite https://helm.sh/docs/helm/helm_dependency/"
+        success_msg: "Failed when there is no dependencies block with dependency_update enabled"
+
+- name: "Test chart with dependencies block and dependency_update true"
+  block:
+    - name: "Test chart with dependencies block and dependency_update true"
+      helm:
+        binary_path: "{{ helm_binary }}"
+        name: test
+        chart_ref: "/tmp/dep-up"
+        chart_version: "{{ chart_source_version | default(omit) }}"
+        namespace: "{{ helm_namespace }}"
+        dependency_update: true
+        create_namespace: yes
+      register: release
+
+    - name: "Check if the subchart exist in chart"
+      stat:
+        path: "/tmp/dep-up/Chart.lock"
+      register: stat_result
+
+    - assert:
+        that:
+          - stat_result.stat.exists
+        success_msg: "subchart exist in the chart directory"
+        fail_msg: "subchart not exist in the charts directory"
+
+# Clean
+- name: "Clean"
```

Comment:
This needs to be in an `always` block. The best way to do this will be to put everything in this file except this block into a single block and then put this part into an `always` block at the bottom. Look at tests_helm_diff.yml for an example.